ndc-models/tests/json_schema/mutation_request.jsonschema

```diff
@@ -141,19 +141,9 @@
               ]
             },
             "arguments": {
-              "type": [
-                "object",
-                "null"
-              ],
+              "type": "object",
```

Comment:
This might be an issue - looks like `skip_serializing_if` doesn't work with `jsonschema`.

Reply:
Oh. I'll see if @BenoitRanque knows a work around.

Reply:
This is working as intended: `arguments` is [not a required field](https://github.com/hasura/ndc-spec/pull/139/files/7036c49c5941c88f2ab7ddefa9061455e6b4e667..ec562f57fb4bd073d480a3a7c26ff66f79487887#diff-9af8bfb06b0046ccdbcbe174bd4ee4d93320b43dc9ee9f8ce589d633858a0092L119-L120).

Do we additionally want to allow `arguments` to be present, but `null`?
That would leave 3 valid empty states: no `argument` field, `arguments` set to `null`, and `arguments` set to `{}`

Reply:
@paf31 ?

Reply:
`schemars` picks up `#[serde(default)]` and makes the field optional. It's true it does not handle `skip_serializing_if`.

So, the schema validates either omitting the `arguments` key, or an object value.
Serde will additionally deserialize from a null value: both `null` and omitting the key are treated as the value being absent, and we fall back to the default value (in out case an empty map).

I do grant this is technically a bug in `schemars`. It looks like they [special cased](https://github.com/GREsau/schemars/issues/11) `Option` to be both nullable and optional.

I'd still argue to keep it this way, as the schema is still consistent with the serialized format.
We just happen to also be able to deserialize from misshapen payloads: null or empty arguments.

Reply:
I must correct myself. `#[serde(default)]` seems to be special cased for `Option` and will happily serialize either null or missing fields to the default. But it will error for other types. So the schemars behavior is correct.

So with the annotations above, `"arguments": null` is not valid, and the schema reflects that.

Do we absolutely need to be able to serialize from `"arguments": null` ?

Reply:
Looks good to me. The tests confirm it's backwards compatible, which is the main thing.